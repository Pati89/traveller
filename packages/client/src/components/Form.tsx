import type { FC } from 'react'
import { FormControl, Input, InputRightElement, IconButton } from '@chakra-ui/react'
import { Search2Icon } from '@chakra-ui/icons'

type Form = {
  onSubmit: (e: any) => Promise<void>
  input: string
  onChange: (e: any) => void
}

export const Form: FC<Form> = ({ onSubmit, input, onChange }) => {
  return (
    <form onSubmit={onSubmit} data-testid="form">
      <FormControl>
        {/* here we can use AsyncSelect component for better user experience */}
        <Input id="search" placeholder="Type city" value={input} onChange={onChange} />
        <InputRightElement
          children={<IconButton aria-label="submit" type="submit" icon={<Search2Icon />} data-testid="submit-button" />}
        />
      </FormControl>
    </form>
  )
}
