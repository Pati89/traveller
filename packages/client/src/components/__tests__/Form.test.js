import { Form } from '../Form'
import { render, screen, fireEvent } from '@testing-library/react'

const props = {
  onSubmit: () => {},
  input: 'test',
  onChange: () => {},
}

describe('Form', () => {
  test('render a Form', () => {
    render(<Form {...props} />)
    const form = screen.getByTestId('form')
    expect(form).toBeTruthy()
  })

  test('render a submit button', () => {
    render(<Form {...props} />)
    const button = screen.getByTestId('submit-button')
    expect(button).toBeTruthy()
  })

  test('should trigger onSubmit function', () => {
    const spy = jest.fn(e => e.preventDefault())
    render(<Form {...props} onSubmit={spy} />)
    const button = screen.getByTestId('submit-button')
    expect(button).toBeTruthy()
    fireEvent.click(button)
    expect(spy).toHaveBeenCalled()
  })

  test('render a Form match snapshot', () => {
    render(<Form {...props} />)

    const form = screen.getByTestId('form')
    expect(form).toMatchSnapshot()
  })
})
