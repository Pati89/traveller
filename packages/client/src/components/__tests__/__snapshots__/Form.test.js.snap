// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render a Form match snapshot 1`] = `
<form
  data-testid="form"
>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <input
      class="chakra-input css-0"
      id="search"
      placeholder="Type city"
      value="test"
    />
    <div
      class="chakra-input__right-element css-dp9r5f"
    >
      <button
        aria-label="submit"
        class="chakra-button css-198bfjk"
        data-testid="submit-button"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</form>
`;
